screens/recipes/NewRecipeScreen.js

```diff
@@ -1,61 +1,61 @@
-import { useStoreActions } from 'easy-peasy';
-import React, { useState } from 'react';
-import { View, Text, ScrollView, TextInput, Alert } from 'react-native';
+import { useStoreActions, useStoreState } from 'easy-peasy';
+import React, { useState, useEffect } from 'react';
+import { View, Text, ScrollView, TextInput } from 'react-native';
 import { TouchableOpacity } from 'react-native-gesture-handler';
 import colors from '../../styles/appColors';
 import styles from '../../styles/Recipes/newRecipe';
 import RecipeSteps from './RecipeStepsScreen';
+// import RecipeIngredients from './RecipeIngredientsScreen';
+import IngredientRow from '../../components/recipeEditIngredientRow';
 import calculateRecipePrice from '../../utils/calculateRecipePrice';
 
-/* eslint max-statements: [2, 20] */
+/* eslint max-statements: [2, 30] */
 function FormRecipe(props) {
   const { navigation, route } = props;
-  const recipe = route ? route.params : null;
+  const recipe = (route.params && route.params.recipe) ? route.params.recipe : null;
+  const setLoadRecipes = useStoreActions((actions) => actions.setLoadRecipes);
 
   const [recipeName, setRecipeName] = useState(recipe ? recipe.attributes.name : '');
   const [recipeTime, setRecipeTime] = useState(recipe ? recipe.attributes.cook_minutes.toString() : '');
   const [recipePortions, setRecipePortions] = useState(recipe ? recipe.attributes.portions.toString() : '');
-  const [recipeSteps, setRecipeSteps] = useState(recipe ? recipe.attributes.steps.data : []);
-  const [recipePrice, setRecipePrice] = useState(recipe ? calculateRecipePrice(recipe) : 0);
+  const [recipeSteps, setRecipeSteps] = useState(
+    recipe ? JSON.parse(JSON.stringify(recipe.attributes.steps.data)) : []);
 
+  const [recipePrice, setRecipePrice] = useState(recipe ? calculateRecipePrice(recipe) : 0);
   const createRecipe = useStoreActions((actions) => actions.createRecipe);
   const editRecipe = useStoreActions((actions) => actions.editRecipe);
+  const [recipeIngredients, setRecipeIngredients] = useState([]);
   const createRecipeStep = useStoreActions((actions) => actions.createRecipeStep);
   const editRecipeStep = useStoreActions((actions) => actions.editRecipeStep);
   const deleteRecipeStep = useStoreActions((actions) => actions.deleteRecipeStep);
 
   function stepsActions(recipeId) {
     const promises = [];
-    let k = 0;
     for (let i = 0; i < recipeSteps.length; i++) {
```

Comment:
acá haría también un forEach para evitar estar accediendo todo el rato a arreglo[i]. Con forEach se simplifica eso y se vuelve más legible el código. En general en js se evitan siempre los for solos, excepto en casos bieeen particulares. Sé que viene de antes pero si te motivai pa arreglarlo ya que tocaste esta parte del código, bacán

Reply:
Arreglado!